Preserve original mtime when copying static files

inukshuk opened the pull request:
Use the `preserve` option when copying static files using `FileUtils.cp`; this will preserve the file's original mtime. To preserve the original mtime of a file is important, for example, when syncing the _site folder.

mattr- commented:
I'm in favor of preserving mtimes when copying static files from the source to the destination directory.

parkr commented:
I can get onboard with this. Looks like `preserve` also copies *all* the other metadata. Are we cool with this, too?

mattr- commented:
@parkr you're right, `preserve` is too much for what we want.

Looks like we'll need special logic for handling mtimes.

parkr commented:
> Looks like we'll need special logic for handling mtimes.

Something like https://github.com/jekyll/jekyll/pull/370/files#diff-17fc003a02cf3a2fc94e8a51d85f3f4fR215 ?

mattr- commented:
Yup. That should do the trick. :smiley:

parkr commented:
Please apply this & fix the tests:

```diff
diff --git a/lib/jekyll/static_file.rb b/lib/jekyll/static_file.rb
index 6d3c6d2..37f2f9f 100644
--- a/lib/jekyll/static_file.rb
+++ b/lib/jekyll/static_file.rb
@@ -80,7 +80,8 @@ module Jekyll
 
       FileUtils.mkdir_p(File.dirname(dest_path))
       FileUtils.rm(dest_path) if File.exist?(dest_path)
-      FileUtils.cp(path, dest_path, :preserve => true)
+      FileUtils.cp(path, dest_path)
+      File.utime(@@mtimes[path], @@mtimes[path], dest_path)
 
       true
     end

```

inukshuk commented:
I have patched and rebased the PR; also updated the test to test that file is regenerated with the same mtime (after being deleted). Please let me know if you would like me to change anything!

parkr commented:
LGTM. /cc @jekyll/core.

alfredxing commented:
:+1:

parkr commented:
@mattr- @envygeeks Any objections?

parkr commented:
:heart:
